src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/base/Functions.java

```diff
@@ -399,19 +512,46 @@ public static boolean checkBuiltInFunctionExists(FunctionIdentifier identifier)
 
     public static RuntimeIterator getBuiltInFunctionIterator(
             FunctionIdentifier identifier,
-            IteratorMetadata metadata,
-            List<RuntimeIterator> arguments
+            List<RuntimeIterator> arguments,
+            IteratorMetadata metadata
     ) {
-        Class<? extends RuntimeIterator> functionClass = builtInFunctions.get(identifier);
+        BuiltinFunction builtinFunction = builtInFunctions.get(identifier);
+
+        for (int i = 0; i < arguments.size(); i++) {
+            if (builtinFunction.getSignature().getParameterTypes().get(i) != mostGeneralSequenceType) {
+                arguments.set(
+                    i,
+                    new TypePromotionIterator(
+                            "Invalid argument for function " + identifier.getName() + ". ",
+                            arguments.get(i),
+                            builtinFunction.getSignature().getParameterTypes().get(i),
+                            metadata
```

Comment:
Guessing: would passing the metadata of iterator arguments.get(i), rather than that of the parent, solve the location hints, in the tests?

Reply:
Using the arguments.get(i).getMetadata() for arguments TypePromotionIterator and functionCallIterator.getMetadata() for return type TypePromotionIterator indeed solves this, bringing back the right column number. Thank you for your help.

Reply:
Very happy to hear it solved the issue! Thank you for fixing this.